internal/controllers/addon_controller.go

```diff
@@ -42,6 +42,9 @@ func (r *AddonReconciler) Reconcile(
 	addon := &addonsv1alpha1.Addon{}
 	err := r.Get(ctx, req.NamespacedName, addon)
 	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			log.Error(err, "unable to fetch Addon")
+		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
There is already a `client.IgnoreNotFound()` present, and controller-runtime will already log any error returned by this reconcile function. So I would rather keep as is.

Reply:
did not know it logged by default. fixed